browsergym/core/src/browsergym/core/env.py

```diff
@@ -390,6 +399,10 @@ def report_infeasible_instructions(reason: str):
             self.chat.add_message(role="infeasible", msg=reason)
             self.infeasible_message_received = True
 
+        if hasattr(self.task, "webcanvas"):
+            logger.debug(f"Initiating  webcanvas task event listen")
+            self._event_listener()
```

Comment:
Same here, this can be done from within the GenericWebCanvasTask class, during `task.setup(page)`.